contentreview-impl/impl/src/java/org/sakaiproject/contentreview/impl/turnitin/TurnitinReviewServiceImpl.java

```diff
@@ -3425,14 +3344,49 @@ private String switchLTIError(int result, String method){
 				return method + ": generic LTI error";
 		}
 	}
-	
-	private void processError(ContentReviewItem item, Long status, String error){//TODO resto de llamadas
-		item.setStatus(status);
-		if(error != null){
+
+	private void processError( ContentReviewItem item, Long status, String error, Integer errorCode )
+	{
+		if( status == null )
+		{
+			IllegalArgumentException ex = new IllegalArgumentException( "Status is null; you must supply a valid status to update when calling processError()" );
+			log.warn( ex );
```

Comment:
I can remove the `log.warn( ex )` here if you think that's more reasonable. In actually, I'm just trying to enforce that a valid `status` is passed to this method.

With regards to the unlocking, I think you are correct 20 of 21 places where this method is called in inside the `processQueue()` method, which does lock the objects. I think your suggestion of wrapping the contents of `processError()` in a `try/finally` where we do a `releaseLock()` in the finally should cover us here.